### PR TITLE
feat-basic authentication

### PR DESCRIPTION
**Pull request by prathmeshjagtap**
Created basic authentication for Login and Signup 

you can not go to cart and Wishlist unless you are logged in

## Files to  Check 
 
Do check basic auth context  , auth reducer ,login form , signUp form , navbar 

- [ ] src/frontend/contexts/auth-context.js
- [ ] src/frontend/pages/login/LoginForm.jsx
- [ ] src/frontend/pages/signup/SignupForm.jsx
- [ ] src/frontend/components/navbar/Navbar.jsx

## What not to check

Have not done too much error Handling 

## To do testing 

Email: test@gmail.com
Password : test@1234

### CODE REVIEW

File: src/frontend/contexts/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const authContext = createContext(null);
+
+const useAuthContext = () => useContext(authContext);
+
+function AuthProvider({ children }) {
+	const localstoragetoken = localStorage.getItem("token");
+
+	const authReducer = (state, action) => {
+		switch (action.type) {
+			case authActions.FIRST_NAME:
```
Comment (SukanyaSen530) [inline src/frontend/contexts/auth-context.js]: first name, last name, email and password can be stored as an object.

**Review by SukanyaSen530 — APPROVED** [review]
Everything else look's great :)

File: src/index.js [hunk 2]
```diff
@@ -4,20 +4,26 @@ import "./index.css";
...
 ReactDOM.render(
 	<React.StrictMode>
-		<BrowserRouter>
-			<DataProvider>
-				<FilterProvider>
-					<App />
-				</FilterProvider>
-			</DataProvider>
-		</BrowserRouter>
+		<AuthProvider>
```
Comment (vijaytembugade) [inline src/index.js:18]: BrowerRouter > AuthProvider, Any particular reason?

**Review by vijaytembugade — APPROVED** [review]
Good work prathamesh.

File: src/frontend/contexts/auth-context.js (hunk 1 above)
Comment (prathmeshjagtap) [reply src/frontend/contexts/auth-context.js]: Yes added

File: src/index.js (hunk 2 above)
Comment (prathmeshjagtap) [reply src/index.js:18]: No reason as such.

**Pull request merged**